src/main/groovy/nextflow/executor/SimpleFileCopyStrategy.groovy

```diff
@@ -171,6 +172,24 @@ class SimpleFileCopyStrategy implements ScriptFileCopyStrategy {
         if( mode == 'symlink' || !mode )
             return "ln -s ${Escape.path(source)} ${Escape.path(target)}"
 
+        if( mode == 'symlinkRelative' ) {
+
+            // Symlink from the working directory to the original location.
+            // GNU ln has the '-r' flag, but BSD ln doesn't, so we have to manually
+            // resolve the relative path
+
+            def targetPath = Paths.get(target)
+            if( ! targetPath.isAbsolute() && workDir != null)
+                targetPath = workDir.resolve(targetPath)
+
+            def sourceRelative = source
+            if( targetPath.isAbsolute() )
+                sourceRelative = targetPath.getParent()
```

Comment:
In which case the `target` is an absolute path? In my understanding it can only be a relative file path.

Reply:
Also please add unit tests for this case. Have a look at [SimpleFileCopyStrategyTest](https://github.com/nextflow-io/nextflow/blob/daded334b2478596d9821fb9049f91e05fc22699/src/test/groovy/nextflow/executor/SimpleFileCopyStrategyTest.groovy#L113)

Reply:
The input `target` value is normally relative, but if `workDir` is defined (as is the case normally), then the new code above will ensure both `targetPath` and `sourcePath` are absolute paths, resolved (if required) relative to `workDir`. If `workDir` is not defined, input relative paths cannot be resolved to absolute paths. Therefore, if `workDir` is defined, or both `source` and `target` are provided as absolute paths, then a relative path can be inferred. Otherwise, the original input paths are used as-is.

I've added a comment explaining this, and added unit tests for the new functionality covering the 'normal' cases as well as all unusual combinations. The tests are passing when I run them locally, but seem to be failing on Travis due to S3 "Access Denied" errors.

Please let me know what you think.

Reply:
I think the target name can be just ignored because the point is the source file path that needs to be relativised. Also you can assume that `workDir` is always defined (it cannot be `null` because it's the task work directory. 

Now basically there are two main conditions: 
1) the source file is the output produced by an upstream task, therefore it must be a path which partially overlap with the task work directory, eg: 
  source path: `/home/foo/work/aa/bbb/file.txt`
  workdir path: `/home/foo/work/xx/yyy`
This needs to be relativised. 

2) the source file does not share a parent path with the work dir. This should not be relativised. 

Since `workDir` is a path you should be able to do something like:

```
if( source.startsWith( workDir.parent.parent.toString() ) ) {
  // relativise 
}
else {
  // absolute
}

Reply:
* The target name used in the returned `ln` command is always the same as the input value. However, the target's context needs to be considered because the relative path is with respect to the target file's directory. This is important when the target contains a subdirectory, and not just a filename, as is the case with some of the existing unit tests.
    
    For example, If `workDir` is `/some/dir`,  `source` is `/some/foo/bar.txt` and `target` is `wibble/wobble.txt`, you need the returned command to be

        ln -s ../../foo/bar.txt wibble/wobble.txt

    If you ignore the `target`'s value, you would have

        ln -s ../foo/bar.txt wibble/wobble.txt

     which would not work.

* For the existing stage-in command unit tests, `workDir` is null. To prevent potential problems in the future, it would be safer to be able to handle this situation gracefully. I felt there were two sensible options: 

    1. an exception stating that `symlinkRelative` is not compatible with a `null` `workDir`, or
    1. using the input values without modification, as done by the `symlink` mode.

    I thought the latter would be easier to work with.

* Even if the `source` and `target`/`workDir` do not share anything beyond the root volume, they will still have relative path.
    * E.g.: the path from `/home/user/work/in.txt` to `/data/static/some_file.txt` would be `../../../data/static/some_file.txt`
    * This is actually important for us, as some tasks we run mount the host root volume under `/host`, and then try to find the input file (`in.txt` in the above example). If an absolute symlink is used, it won't find the file (as `/data` is present on the host, and found under `/host/data` in the guest machine).

Reply:
Hi Paolo, I might have misunderstood you, but the relative path is from the target file location to the source file location. Whether or not they are in the same directory, they will have a relative path. For example, if source is `/data/file.txt` and target is `/home/user/abc.txt`, the relative path from target to source is `../../data/file.txt`.

The only time this can't be inferred is if the location of either path is ambiguous, which only happens when `workDir` is undefined and either path is not absolute, which - as you said - won't occur normally as `workDir` is always defined.

Reply:
You are right, OK. Therefore, you can assume 1) `workDir != null` and an absolute path, 2) targetPath is a relative path.

Reply:
Great. The code as submitted works for that circumstance and has tests covering it. It also handles and tests the other cases we discussed (just in case they are encountered) with minimal additional code.

Please let me know if it is suitable or requires any changes.

Reply:
Sorry, but I'm still not understanding why there are checks like the following 

```
targetPath.isAbsolute() && workDir != null
```

My previous comment
 
> Therefore, you can assume 1) workDir != null and an absolute path, 2) targetPath is a relative path

Reply:
Those assumptions should be correct, but we cannot guarantee that will always be the case. If for some unforeseen reason these assumptions are not met, the code can behave in one of three ways:

1. Detect the error and report a meaningful message,
2. Fallback to a sensible alternative response, or
3. Throw an exception because we call `resolve` on a `null` `workDir` object.

I chose option 2, since it required very little extra code. I can remove the extra line of code if you like, but I don't see any benefit from doing so.

Reply:
Thanks for the CLA. Regarding the other point since it is supposed to be an illegal condition, I would prefer to throw an error. For example adding an assert condition at the beginning, something like: 

```
assert workDir, "Task work dir path must be define"
assert !targetPath.startsWith('/'), "Target path must be relative"

Reply:
Very well. I have made the changes as discussed and updated/added tests to verify this behaviour. This PR is ready for re-review.